### PR TITLE
bpf: implement unconnected udp based host lb

### PR DESCRIPTION
**Pull request by borkmann**
This gets nslookup/dig working for unconnected UDP over IPv4/IPv6.

For details, see commit message.

(Requires kernel fix and iproute2/bpftool modification before we can push to Cilium repo.)

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8100)

### CODE REVIEW

File: bpf/bpf_sock.c [hunk 1]
```diff
@@ -95,6 +106,105 @@ int sock4_xlate(struct bpf_sock_addr *ctx)
...
+{
+	struct ipv4_revnat_entry *rval;
+	struct ipv4_revnat_tuple rkey = {
+		.cookie		= sock_cookie_or_zero(ctx),
+		.address	= ctx->user_ip4,
+		.port		= ctx_get_port(ctx),
+	};
+
+	rval = map_lookup_elem(&cilium_cgroups_revnat4, &rkey);
+	if (!rval) {
```
Comment (vadorovsky) [inline bpf/bpf_sock.c]: Maybe

```
if (rval) {
    struct lb4_service_v2 *svc;
    ...
}

return RECVMSG_PROCEED;
```
?

File: bpf/bpf_sock.c [hunk 2]
```diff
@@ -150,6 +259,106 @@ int sock6_xlate(struct bpf_sock_addr *ctx)
...
+{
+	struct ipv6_revnat_tuple rkey = {};
+	struct ipv6_revnat_entry *rval;
+
+	rkey.cookie = sock_cookie_or_zero(ctx);
+	rkey.port = ctx_get_port(ctx);
+	ctx_get_v6_address(ctx, &rkey.address);
+
+	rval = map_lookup_elem(&cilium_cgroups_revnat6, &rkey);
+	if (!rval) {
```
Comment (vadorovsky) [inline bpf/bpf_sock.c]: Maybe

```
if (rval) {
    struct lb6_service_v2 *svc;
    ...
}

return RECVMSG_PROCEED;
```
?

**Review by vadorovsky — COMMENTED** [review]
One minor nit.

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
Seems that the CI is hitting https://github.com/moby/moby/issues/34672.

**Comment by borkmann** [comment]
Merged via #8375. Closing this one.

**Pull request closed without merging**